### PR TITLE
added check for staging environments

### PR DESCRIPTION
**Pull request by UVLabs**

### CODE REVIEW

File: includes/admin/services/class-rop-facebook-service.php [hunk 1]
```diff
@@ -458,20 +458,26 @@ public function share( $post_details, $args = array() ) {
...
-		if ( $this->try_post( $new_post, $args['id'], $args['access_token'] ) ) {
-			$this->logger->alert_success(
-				sprintf(
-					'Successfully shared %s to %s on %s ',
-					get_the_title( $post_details['post_id'] ),
-					$args['user'],
-					$post_details['service']
-				)
-			);
+		$rop_staging_status = new Rop();
```
Comment (preda-bogdan) [inline includes/admin/services/class-rop-facebook-service.php]: Don't instantiate the main class for the `rop_site_is_staging()` method. Instead either move the method to a utility class that can be instantiated multiple times or inject the result of the method during service factory.
I would do a `protected $is_staging` variable in the abstract service class that is instantiated in the `__construct()` with the value from the method in the utility class.

File: includes/class-rop.php [hunk 2]
```diff
@@ -163,38 +163,37 @@ public function get_version() {
...
 
 	/**
-	 * detects if is a staging environment
+	 * Detects if is a staging environment
 	 *
-	 * @since     8.0.3
+	 * @since     8.0.4
 	 * @return    bool   true/false
 	 */
+	public function rop_site_is_staging() {
```
Comment (preda-bogdan) [inline includes/class-rop.php]: Maybe move this method to a utility class.

**Comment by UVLabs** [comment]
Changes made @preda-bogdan

**Comment by preda-bogdan** [comment]
@UVLabs Looks good now. It makes more sense to have it in the abstract class don't you think?

**Comment by preda-bogdan** [comment]
@UVLabs I have fixed the conflict for you. 
You can read more about conflicts here for further reference https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/

**Comment by selul** [comment]
@UVLabs do you think you can tackle the Vue templates and add in the sidebar a notice when the user is in the staging mode? Under the current sidebar boxes we can add a simple notice box letting them know that they are in staging mode and no posts will be sent to social networks.

**Comment by UVLabs** [comment]
@selul sure I'll check it out soon and add to this pull

**Pull request merged**